src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -96,6 +98,54 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
             codeDocument.SetDocumentIntermediateNode(document);
         }
 
+        private void InheritImportedDirectives(DocumentIntermediateNode document, string mainDocumentFilePath)
+        {
+            var visitor = new DirectiveVisitor();
+            visitor.VisitDocument(document);
+
+            var seenDirectives = new HashSet<DirectiveDescriptor>();
+            for (var i = visitor.Directives.Count - 1; i >= 0; i--)
+            {
+                var reference = visitor.Directives[i];
+                var directive = (DirectiveIntermediateNode)reference.Node;
+                var descriptor = directive.Descriptor;
+                var seenDirective = !seenDirectives.Add(descriptor);
+                var imported = directive.Source.HasValue && directive.Source.Value.FilePath != mainDocumentFilePath;
```

Comment:
We don't need to use file paths here. We have the code document and we know what are imports and what aren't.